crates/iota-data-ingestion-core/src/reader.rs

```diff
@@ -29,7 +29,13 @@ use crate::{
     executor::MAX_CHECKPOINTS_IN_PROGRESS,
 };
 
-type CheckpointResult = IngestionResult<(CheckpointData, usize)>;
+// A thread-safe reference-counted wrapper around [`CheckpointData`].
+/// This type alias provides a convenient way to share [`CheckpointData`] across
+/// multiple threads using [`Arc`]. Multiple threads can hold references to the
+/// same [`CheckpointData`] instance, and it will be automatically cleaned up
+/// when the last reference is dropped.
+pub type SharedCheckpointData = Arc<CheckpointData>;
```

Comment:
I wonder if the type alias is warranted in this context. Having `Arc<CheckpointData>` has more clear semantics than `SharedCheckpointData`. 

So for example when we read a function implementation that takes `SharedCheckpointData` and see methods of `CheckpointData` called within due to `Deref`, we don't have to trace back its origin in the type alias definition, but just rely on `Arc<CheckpointData>` in the function signature.

I therefore think it would actually be more readable without the alias.

Reply:
[fixed](https://github.com/iotaledger/iota/pull/5336/commits/78695328c8e433bb8ef3e05acf634d89057c2d54)

regarding the optimization, the Arc is useful in this part of code: https://github.com/iotaledger/iota/blob/develop/crates/iota-data-ingestion-core/src/executor.rs#L130

My understanding is that previously for every worker pool the `CheckpointData` was cloned, which was expensive, the more worker pools registered the more the memory was increasing.

Reply:
Got it, makes sense, thank you!